### PR TITLE
Add official Pi-hole WireGuard guide

### PR DESCRIPTION
**Pull request by DL6ER**
### This adds a guide how to install WireGuard on your Pi-hole.

You may ask: Why another guide when there are already quite a few on the web?

The answer is manifold:

1. The vast majority of online guides only sets up an IPv4 tunnel. **The Pi-hole guide gives you IPv4 and IPv6 through the same tunnel.**
2. This guide is **build around a DNS server being in the center of the tunnel**. This ensures the settings are fine-tuned so no DNS leakage happens.
3. Some guides show you how to make your internal network clients available through the tunnel, sometimes, they tell you how to route your entire traffic over the Internet. **In our guide, we show you all the options and you decide exactly what you want**. Everything in one place means a similar writing style and expectation on what you did before. This makes it easier for you to set things up.
4. The guide has been written to include **support for old kernels** for which you have to compile the kernel module yourself. No searching around on the web for instructions. Everything is already there.
5. **Concerning security, the Pi-hole WireGuard guide pulls all the stops** while still being very simple to set up. This starts in explaining how to generate safe keys and does not stop at using an additional layer of security by using simple per-connection pre-shared keys (PSK).

### CODE REVIEW

**Review by dschaper — APPROVED** [review]

File: docs/guides/wireguard/client.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+Restart your server to load the new client config:
+
+```bash
+systemctl restart wg-quick@wg0
+```
+
+After a restart, the server file should look like:
+
+```toml
```
Comment (XhmikosR) [inline docs/guides/wireguard/client.md:83]: Are you sure `toml` is right? I see the GitHub parser produces wrong output.

File: docs/guides/wireguard/client.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ For each new client, the following steps must be taken. For the sake of simplici
 
 <!-- markdownlint-disable code-block-style -->
 ??? info "All commands described below at once"
-    ``` bash
+    ```bash
```
Comment (dschaper) [inline docs/guides/wireguard/client.md:9]: Does this matter really? Spec is that the infostring is the first word after the triple ticks. Extra space "shouldn't" matter?

Comment (XhmikosR) [reply docs/guides/wireguard/client.md:9]: It doesn't matter, but it's inconsistent with the codebase; we don't use spaces anywhere else apart from this branch.

File: docs/guides/wireguard/client.md (hunk 1 above)
Comment (dschaper) [reply docs/guides/wireguard/client.md:83]: `toml` looks odd to me actually and I don't think there is a `toml` format? Could do with nothing and just leave it unstyled. 

http://www.rubycoloredglasses.com/2013/04/languages-supported-by-github-flavored-markdown/ shows the languages, but I don't think we're limited to GFM, we could use any lang spec supported by plugin.

Comment (dschaper) [reply docs/guides/wireguard/client.md:83]: `bash` displays a nice output. Comments are noted and everything else stays plain. (Except that bool)
Screenshot 2020-10-18 224323

Comment (DL6ER) [reply docs/guides/wireguard/client.md:83]: Yes, it's true that GitHub is producing incorrect output, however, this shouldn't stop us from using a formatter which is supported by what matters in the end - `mkdocs`. I'm fine with plain if you prefer this, but still think color is nice for configuration files on long pages.

If you could point me to the source code of the highlighter used in `mkdocs`, I can make a PR there to add `wireguard.conf` as new format with 100% proper highlighting

Comment (dschaper) [reply docs/guides/wireguard/client.md:83]: What you see in the netlify preview link is what you'll see on the actual deployed site.

Comment (XhmikosR) [reply docs/guides/wireguard/client.md:83]: I'd go with whatever produces better output. Toml does not seem to be close to what we want, so maybe what @dschaper suggested is better compared to no highlighting?

It's a tricky situation for sure, hence why, personally, I go with plan text when there isn't a 100% language I can use.

**Comment by XhmikosR** [comment]
@DL6ER it seems there are more `toml` instances.

**Comment by XhmikosR** [comment]
I think this is good to go, apart from the wrong highlighting language used in some places (`bash`). Personally I don't use Wireguard (yet) so I can't confirm the details but most things look good to me apart from highlighting.

**Comment by PromoFaux** [comment]
A slight nitpick, but perhaps we should rearrange these to be alphabetical?

image

Else, why is Wireguard at the top(ish)?

**Comment by DL6ER** [comment]
It is indeed top(ish) because I consider it to be one of the more-often looked up guides. It should at least come above OpenVPN as Wireguard should be used instead of the the former for new installed. I know that this is certainly some level of personal opinion, but alphabetical doesn't seem right here, either. Just put "Configure" in front to bring your item further up.

**Comment by yubiuser** [comment]
`Endpoint = [your public IP or domain]:47111`

You may want to add a hint for beginners that they could use a DDNS service on their router in case they don't have a stable public IP.

**Comment by yubiuser** [comment]
Add: is there a reason you chose this port?
Usually I've seen 51820, which is also used here
https://www.wireguard.com/quickstart/

**Comment by DL6ER** [comment]
> Usually I've seen 51820, which is also used here

It's not necessary to stick to a specific port as only devices that know the configuration will connect. It may be good idea to scatter services across ports and not always put Wireguard at the same port.

**Comment by yubiuser** [comment]
I've once had the issue after a reboot that the wg interface wasn't up before the pihole-FTL service was up. So pihole didn't bind to that interface. This issue led to the delay startup option
https://github.com/pi-hole/FTL/pull/716

Would this be something for the troubleshoot section?

**Comment by DL6ER** [comment]
@yubiuser Thanks for your suggestions. I just pushed them to this branch.

**Comment by DL6ER** [comment]
Rebased on latest `master`

**Review by PromoFaux — APPROVED** [review]

**Pull request merged**